src/IIIFPresentation/API/Features/Storage/Requests/CreateCollection.cs

```diff
@@ -75,20 +76,50 @@ public async Task<ModifyEntityResult<PresentationCollection, ModifyCollectionTyp
             Modified = dateCreated,
             CreatedBy = Authorizer.GetUser(),
             CustomerId = request.CustomerId,
+            Tags = request.Collection.Tags,
             IsPublic = request.Collection.Behavior.IsPublic(),
             IsStorageCollection = request.Collection.Behavior.IsStorageCollection(),
-            Label = request.Collection.Label,
-            Parent = parentCollection.Id,
+            Label = request.Collection.Label
+        };
+
+        var hierarchy = new Hierarchy
+        {
+            CollectionId = id,
+            Type = request.Collection.Behavior.IsStorageCollection()
+                ? ResourceType.StorageCollection
+                : ResourceType.IIIFCollection,
             Slug = request.Collection.Slug,
-            Thumbnail = request.Collection.Thumbnail,
-            Tags = request.Collection.Tags,
-            ItemsOrder = request.Collection.ItemsOrder
+            CustomerId = request.CustomerId,
+            Canonical = true,
+            ItemsOrder = request.Collection.ItemsOrder,
+            Parent = request.Collection.Parent,
+            Public = request.Collection.Behavior.IsPublic()
         };
         
+        string? convertedIIIFCollection = null;
+
+        if (!request.Collection.Behavior.IsStorageCollection())
```

Comment:
`request.Collection.Behavior.IsStorageCollection()` isn't particularly expensive but it's called 4 times in this method. Can it be simplified - either the result stored or `collection.IsStorageCollection` used as that's set first?